test/unit/pages/patientdata.test.js

```diff
@@ -85,152 +98,165 @@ describe('PatientData', function () {
-        expect(y).to.be.ok;
+        wrapper.setProps({ fetchingPatientData: true });
+        expect(loader().props().show).to.be.true;
       });
 
       it('should render the loading message and image when both fetchingPatient and fetchingPatientData are true', function() {
-        var props = {
-          fetchingPatient: true,
-          fetchingPatientData: true
-        };
-
-        // Try out using the spread props syntax in JSX
-        var elem = TestUtils.renderIntoDocument(<PatientData {...props}/>);
-        expect(elem).to.be.ok;
-
-        var x = TestUtils.findRenderedDOMComponentWithClass(elem, 'patient-data-loading-image');
-        expect(x).to.be.ok;
+        expect(loader().length).to.equal(1);
+        expect(loader().props().show).to.be.false;
 
-        var y = TestUtils.findRenderedDOMComponentWithClass(elem, 'patient-data-loading-message');
-        expect(y).to.be.ok;
+        wrapper.setProps({
+          fetchingPatient: true,
+          fetchingPatientData: true,
+        });
+        expect(loader().props().show).to.be.true;
       });
 
       it('should still render the loading message and image when fetching is done but data is being processed', function() {
         var props = {
           fetchingPatient: false,
-          fetchingPatientData: false
+          fetchingPatientData: false,
         };
 
-        // Try out using the spread props syntax in JSX
-        var elem = TestUtils.renderIntoDocument(<PatientData {...props}/>);
-        expect(elem).to.be.ok;
+        var state = {
+          processingData: true,
+        };
 
-        var x = TestUtils.findRenderedDOMComponentWithClass(elem, 'patient-data-loading-image');
-        expect(x).to.be.ok;
+        expect(loader().length).to.equal(1);
+        expect(loader().props().show).to.be.false;
 
-        var y = TestUtils.findRenderedDOMComponentWithClass(elem, 'patient-data-loading-message');
-        expect(y).to.be.ok;
+        wrapper.setProps(props);
+        wrapper.setState(state);
+        expect(loader().props().show).to.be.true;
       });
 
       // this is THE REGRESSION TEST for the "data mismatch" bug
       it('should continue to render the loading message when data has been fetched for someone else but not for current patient', () => {
         var props = {
           currentPatientInViewId: 41,
           fetchingPatient: false,
-          fetchingPatientData: false
+          fetchingPatientData: true,
         };
 
-        // Try out using the spread props syntax in JSX
-        var elem = TestUtils.renderIntoDocument(<PatientData {...props}/>);
-        // bypass the actual processing function since that's not what we're testing here!
-        elem.doProcessing = sinon.spy();
-        elem.componentWillReceiveProps({
+        wrapper.setProps(props);
+
+        expect(loader().length).to.equal(1);
+        expect(loader().props().show).to.be.true;
+
+        wrapper.setProps({
```

Comment:
Hmm. I'm looking into this one... the test breaks if I set it to `false`.  I would expect this, though, given the current logic that determines when the loader should show.

This was basically just a port of the existing test to `enzyme`.  I'm not sure _exactly_ the regression that was being addressed here, but it seems pretty obvious what the intent was.

I'm thinking that this test could likely be removed.  I assume that at some point, it was hiding the loading even when `fetchingPatientData` was still `true` due data having already been fetched for a different user.

I actually had thought about removing it, as I couldn't see how this would occur with the current logic, but was hesitant to remove it because it was meant to future-proof against an obscure issue that had likely made it's way to production at some point.

What do you think @krystophv?

Reply:
I'm almost certain that it was due to a race condition where a clinician would switch between viewing two accounts quickly (or not so quickly since data takes so long to load) and would end up in a situation where the data displayed was for an account that wasn't supposed to be currently active in the display (since the redux actions for fetching would finish for one set while another had already started: fetch1, fetch2, success1, success2 - it's to prevent viewing any of the data from success1). I can take another glance, but perhaps the new logic prevents this situation entirely?